packages/trader/src/Stores/Modules/Trading/trade-store.ts

```diff
@@ -42,6 +42,119 @@ import BaseStore from '../../base-store';
 import { ChartBarrierStore } from '../SmartChart/chart-barrier-store';
 import debounce from 'lodash.debounce';
 import { setLimitOrderBarriers } from './Helpers/limit-orders';
+// import { useP2PNotificationCount } from '@deriv/hooks';
+import type { TCoreStores } from '@deriv/stores/types';
+import { ActiveSymbols, BuyContractResponse, PriceProposalRequest } from '@deriv/api-types';
+
+type TBarriers = Array<
+    ChartBarrierStore & {
+        hideOffscreenBarrier?: boolean;
+        isSingleBarrier?: boolean;
+    }
+>;
+type TChartLayout = {
+    adj: boolean;
+    aggregationType: string;
+    animation?: boolean;
+    candleWidth: number;
+    chartScale: string;
+    chartType: string;
+    crosshair: number;
+    extended: boolean;
+    flipped: boolean;
+    interval: number;
+    marketSessions: Partial<Record<string, boolean>>;
+    outliers: boolean;
+    panels: {
+        chart: {
+            chartName: string;
+            display: string;
+            index: number;
+            percent: number;
+            yAxis: {
+                name: string;
+                position: null;
+            };
+            yaxisLHS: string[];
+            yaxisRHS: string[];
+        };
+    };
+    periodicity: number;
+    previousMaxTicks?: number;
+    range: Partial<Record<string, unknown>>;
+    setSpan: Partial<Record<string, unknown>>;
+    studies?: Partial<Record<string, unknown>>;
+    symbols: [
+        {
+            interval: number;
+            periodicity: number;
+            setSpan: Partial<Record<string, unknown>>;
+            symbol: string;
+            symbolObject: ActiveSymbols[number];
+            timeUnit: string;
+        }
+    ];
+    timeUnit: string;
+    volumeUnderlay: boolean;
+};
+type TPrevChartLayout =
+    | (TChartLayout & {
+          isDone?: VoidFunction;
+          is_used?: boolean;
+      })
+    | null;
+type TContractCategoriesList = Partial<
+    Record<
+        | 'Multipliers'
+        | 'Ups & Downs'
+        | 'Highs & Lows'
+        | 'Ins & Outs'
+        | 'Look Backs'
+        | 'Digits'
+        | 'Vanillas'
+        | 'Accumulators',
+        TContractTypesList
+    >
+>;
+type TContractTypesList = {
+    [key: string]: {
+        name: string;
+        categories: TTextValueStrings[];
+    };
+};
+type TDurationMinMax = {
+    [key: string]: { min: number; max: number };
+};
+type TProposalInfo = {
+    [key: string]: ReturnType<typeof getProposalInfo>;
+};
+type TStakeBoundary = Record<
+    string,
+    {
+        min_stake?: string;
+        max_stake?: string;
+    }
+>;
+type TTextValueStrings = {
```

Comment:
Thinking about exporting this to common-prop.type.ts. Seeing it used in more than a few files.